File: Solution/CSharp/169.MajorityElement.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int length = nums.Length;
        int halfLength = length / 2;
        Dictionary<int, int> d = new Dictionary<int, int>();
        int value = 0;
        int num = 0;
        for (int i = 0; i < length; i++) {
            num = nums[i];
            if (d.TryGetValue(num, out value)) {
                d[num] += 1;
            } else {
                d[num] = 1;
            }
        }

        int element = 0;
        foreach (int i in d.Keys) {
            if (d[i] > halfLength) {
                element = i;
            }
        }
        return element;
    }
}

File: Solution/Python/169. Majority Element.py
class Solution:
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return None
        if len(nums) == 1:
            return nums[0]
        a = self.majorityElement(nums[:len(nums) // 2])
        b = self.majorityElement(nums[len(nums) // 2:])
        if a == b:
            return a
        return [b, a][nums.count(a) > len(nums) // 2]


s = Solution()
nums = [7, 7, 5, 7, 5, 1, 5, 7, 5, 5, 7, 7, 7, 7, 7, 7]
res = s.majorityElement(nums)
print(res)
